cli/src/test-utils/test-runner.spec.ts

```diff
@@ -0,0 +1,209 @@
+import nock from "nock";
+import { cleanse } from "../../../lib/src/cleansers/cleanser";
+import { ContractDefinition } from "../../../lib/src/models/definitions";
+import { parse } from "../../../lib/src/parsers/parser";
+import { verify } from "../../../lib/src/verifiers/verifier";
+import { runTest } from "./test-runner";
+
+describe("test runner", () => {
+  const stateUrl = "http://localhost:9988/state";
+  const baseUrl = "http://localhost:9988";
+
+  afterEach(() => {
+    nock.cleanAll();
+  });
+
+  test("no provider states passes", async () => {
+    const contract = parseAndCleanse(
+      "./cli/src/test-utils/test-runner-examples/no-provider-states.ts"
+    );
+
+    const scope = nock(baseUrl)
+      .post("/companies", { name: "My Company", private: true })
+      .reply(
+        201,
+        { name: "My Company" },
+        { Location: `${baseUrl}/companies/abc` }
+      )
+      .post("/state")
+      .query({ action: "teardown" })
+      .reply(200);
+
+    const result = await runTest(contract, stateUrl, baseUrl);
+    expect(scope.isDone()).toEqual(true);
```

Comment:
FYI I think the more common approach is to use `toBe()` for primitive values: https://jestjs.io/docs/en/using-matchers